stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,292 @@
+        warehouse = self.env["stock.warehouse"].search([])
+        if len(warehouse) == 1:
+            return warehouse
+
+    @api.depends("from_picking_type_id.default_location_src_id")
+    def _compute_from_location_src_id(self):
+        for record in self:
+            location = record.from_picking_type_id.default_location_src_id
+            if not location:
+                __, location = self.env["stock.warehouse"]._get_partner_locations()
+            record.from_location_src_id = location
+
+    @api.depends("from_picking_type_id.default_location_dest_id")
+    def _compute_from_location_dest_id(self):
+        for record in self:
+            location = record.from_picking_type_id.default_location_dest_id
+            if not location:
+                location, __ = self.env["stock.warehouse"]._get_partner_locations()
+            record.from_location_dest_id = location
+
+    def _compute_impacted_route_ids(self):
+        for flow in self:
+            rules = self.env["stock.rule"].search(
+                [
+                    ("picking_type_id", "=", flow.from_picking_type_id.id),
+                    ("route_id.flow_id", "=", False),
+                ]
+            )
+            flow.impacted_route_ids = rules.route_id
+
+    @api.constrains(
+        "warehouse_id", "from_picking_type_id", "carrier_ids", "move_domain"
+    )
+    def _constrains_uniq(self):
+        args = [
+            ("warehouse_id", "=", self.warehouse_id.id),
+            ("from_picking_type_id", "=", self.from_picking_type_id.id),
+        ]
+        flows = self.search(args) - self
+        for flow in flows:
+            if (
+                self.carrier_ids & flow.carrier_ids
+                or not self.carrier_ids
+                and not flow.carrier_ids
+            ) and self.move_domain == flow.move_domain:
+                raise UserError(
+                    _("Existing flow '%s' already applies on these kind of moves.")
+                    % flow.name
+                )
+
+    @api.onchange("name")
+    def onchange_name(self):
+        self.sequence_prefix = "".join(
+            [w[0] for w in (self.name or "").split()]
+        ).upper()
+
+    @api.onchange("warehouse_id")
+    def onchange_warehouse_id(self):
+        self.from_picking_type_id = self.warehouse_id.out_type_id
+
+    def _generate_delivery_route(self):
+        self.ensure_one()
+        if self.delivery_route_id:
+            return False
+        # Re-generate the WH configuration with the chosen delivery steps in a
+        # savepoint to get a whole route+rules+picking types configured almost
+        # automatically (to not re-invent the wheel).
+        # Delivery route values are then copied to generate a new one.
+        # Copy some picking types with new sublocations first so the new
+        # delivery route will use them.
+        pick_type, pack_type, out_type = False, False, False
+        out_type = self.warehouse_id.out_type_id.copy(
```

Comment:
The new picking types should be generated only if they are not provided in the configuration on the warehouse flow. If they are generated, they should be accessible on the warehouse flow (like you have the picking types on the tech info of a warehouse)

Reply:
Indeed, this still need to be implemented...